octobot_trading/exchanges/connectors/ccxt_exchange.py

```diff
@@ -301,91 +493,474 @@ async def get_all_currencies_price_ticker(self, **kwargs: dict) -> typing.Option
+                                                    ) -> typing.Optional[dict]:
+        for order in await self.get_open_orders(symbol, check_completeness=check_completeness, **kwargs):
+            if order[ecoc.ID.value] == order_id:
+                return order
+        for order in await self.get_closed_orders(symbol, check_completeness=check_completeness, **kwargs):
+            if order[ecoc.ID.value] == order_id:
+                return order
+        return None  # OrderNotFound
+
+    async def get_order_from_trades(self, order_id, symbol, check_completeness=True):
+        trades = await self.get_my_recent_trades(symbol, check_completeness=check_completeness)
+        # usually the right trade is within the last ones
+        for trade in trades[::-1]:
+            if trade[ecoc.ID.value] == order_id:
+                return trade
         return None  # OrderNotFound
 
-    async def get_all_orders(self, symbol: str = None, since: int = None,
-                             limit: int = None, **kwargs: dict) -> list:
-        if self.client.has['fetchOrders']:
+    async def get_order_private(self, order_id: str, symbol: str = None, check_completeness: bool = True,
+                                **kwargs: dict) -> typing.Optional[dict]:
+        # todo moved from bybit - should be checked
+        # fetchOrder() supports spot markets and linear non-USDC perpetual swap markets only
+        # try to force fetch order with id even on USDC / non-linear markets
+        # from ccxt/async_support/bybit/fetch_order
+        # from ccxt/async_support/bybit/fetch_orders
+        market = self.client.market(symbol)
+        isStop = self.client.safe_value(kwargs, 'stop', False)
+        stopOrderId = self.client.safe_string(kwargs, 'stop_order_id')  # might want to filter by id
+        orderType = self.client.safe_string_lower(kwargs, 'orderType')
+        isConditionalOrder = isStop or (stopOrderId is not None) or (orderType == 'stop' or orderType == 'conditional')
+        params = self.client.omit(kwargs, ['orderType', 'stop', 'orderType'])
+        if stopOrderId is None:
+            if isConditionalOrder:
+                orderKey = 'stop_order_id'
+            else:
+                isUsdcSettled = (market['settle'] == 'USDC')
+                orderKey = 'orderId' if isUsdcSettled else 'order_id'
+            if order_id is not None:  # The user can also use argument params["order_link_id"] and leave self as None
+                params[orderKey] = order_id
+        if market['linear']:
+            method = 'privateGetPrivateLinearStopOrderList' if isConditionalOrder else 'privateGetPrivateLinearOrderList'
+        elif market['future']:
+            method = 'privateGetFuturesPrivateStopOrderList' if isConditionalOrder else 'privateGetFuturesPrivateOrderList'
+        else:
+            # inverse swap
+            method = 'privateGetV2PrivateStopOrderList' if isConditionalOrder else 'privateGetV2PrivateOrderList'
```

Comment:
private ccxt methods are generated from api endpoints. They are generated code that can change from a ccxt version to the other without warning and depend on the exchange api endpoints structure. I think we should keep these in exchanges tentacles

Reply:
yeah makes sense, do you know why that was made for bybit? it seems to work without it